hikari/impl/buckets.py

```diff
@@ -303,6 +327,28 @@ def drip(self) -> None:
         if not self.is_unknown:
             self.remaining -= 1
 
+    def resolve(self, real_bucket_hash: str) -> None:
+        """Resolve an unknown bucket.
+
+        Parameters
+        ----------
+        real_bucket_hash: builtins.str
+            The real bucket hash for this bucket.
+
+        Raises
+        ------
+        builtins.RuntimeError
+            If the hash of the bucket is already known.
+        """
+        if not self.is_unknown:
+            raise RuntimeError("Cannot resolve known bucket")
+
+        self.name = real_bucket_hash
+
+
+def _create_unknown_hash(route: routes.CompiledRoute) -> str:
+    return UNKNOWN_HASH + routes.HASH_SEPARATOR + str(hash(route))
```

Comment:
What about major params?

Reply:
The route already includes the major parameters